### PR TITLE
Support per-test tolerances for ONNX tests

### PR DESCRIPTION
**Pull request by garymm**
Prior to this every test shared the same global tolerances. This meant
that if an ONNX test failed due to a small but acceptable difference in
output, the only alternative was to disable the test entirely.

In op set 17, the DFT operator is being added. Without this change, the
tests for that operator fail because the output is off by about 5e-5.
It's better to keep test coverage for this new op rather than disable
the test entirely.

Also prior to this change, the global tolerances were not shared between
JavaScript, Python and C++ tests. Now they are.

Also update the JSON submodule, which is needed to get it to build with
exceptions disabled.

Unblocks https://github.com/microsoft/onnxruntime/issues/11640.

### CODE REVIEW

File: onnxruntime/test/python/onnx_backend_test_series.py
Comment (garymm) [inline onnxruntime/test/python/onnx_backend_test_series.py:18]: I have no idea why this is causing this check to fail.

File: js/node/test/test-utils.ts
Comment (jcwchen) [inline js/node/test/test-utils.ts:90]: Maybe not relevant to this PR: I am curious what are these numbers (1.0e-4, 1.0e-6) for since it seems to have ATOL_DEFAULT and RTOL_DEFAULT already.

**Review by jcwchen — COMMENTED** [review]
Thanks for proposing per-test tolerances! The document improvement looks good to me. It would be better to get other experts' approvals about the Python/JavaScript part. Going forward with this PR, perhaps in the future we can go through the whole skip list for ONNX node tests and enable some of them with specific tolerances if reasonable.

**Comment by fs-eire** [comment]
please fix the build errors of code formatting and ESLint by running the following commands under <ORT_ROOT>/js/ :
```sh
npm ci
npm run format
npm run lint
```

**Comment by garymm** [comment]
@fs-eire @jcwchen could you please review?

File: js/node/test/test-utils.ts
Comment (fs-eire) [inline js/node/test/test-utils.ts:281]: as "atol_default" and "rtol_default" are special keys in override json, you can directly define them as in the type of OVERRIDES. so the `as` keyword can be removed in the following few lines.

Comment (fs-eire) [reply js/node/test/test-utils.ts:281]: also, could you move this to the top next to BACKEND_TEST_SERIES_FILTERS?

File: cmake/onnxruntime_unittests.cmake
Comment (jcwchen) [inline cmake/onnxruntime_unittests.cmake:860]: Just out of curiosity: Was `nlohmann_json::nlohmann_json` added because updated json submodule?

**Review by jcwchen — APPROVED** [review]
Overall LGTM. It would be great if this PR can get approvals from ORT Core team and ORT Web team respectively.

File: cmake/onnxruntime_unittests.cmake
Comment (garymm) [reply cmake/onnxruntime_unittests.cmake:860]: No, because I need to use it to parse `onnx_backend_test_series_overrides.jsonc` in `onnx_test_runner`. See `main.cc`.

File: js/node/test/test-utils.ts
Comment (garymm) [reply js/node/test/test-utils.ts:281]: Done.

File: winml/test/model/model_tests.cpp
Comment (yuslepukhin) [inline winml/test/model/model_tests.cpp:176]: >doubl [](http://example.com/codeflow?start=2&length=5)

constexpr

**Comment by yuslepukhin** [comment]
LGTM. As soon as checks pass I can sign off.

File: winml/test/model/model_tests.cpp
Comment (garymm) [reply winml/test/model/model_tests.cpp:176]: Sorry I'm not sure what you're asking me to do. Could you please elaborate?

File: onnxruntime/test/onnx/main.cc
Comment (garymm) [reply onnxruntime/test/onnx/main.cc]: Inlined it instead.

**Review by fs-eire — APPROVED** [review]
signed off for typescript files (/js/node/**)

**Comment by garymm** [comment]
@yuslepukhin all checks passed so I'll merge now. If there are any changes you'd like to see, feel free to comment and I'll open another PR.

**Pull request merged**
